### PR TITLE
Better exception handling, settings page fixes, disconnection fix, added welcome screen - lots of cleanups

### PR DESCRIPTION
**Pull request by JimGaleForce**
Better exception handling, settings page fixes (button text, header text), disconnection fix when entering settings page, added welcome screen - lots of cleanups (reorganized code within files / resharper + stylecop - documentation).

All should be cleanup only (moving/reformatting code) except where I commented in git.

### CODE REVIEW

File: Shield/SettingsPage.xaml [hunk 1]
```diff
@@ -30,6 +30,20 @@
...
+            <Setter Property="Margin" Value="0,0,0,-10" />
+            <Setter Property="FontFamily" Value="Segoe UI Light" />
+            <Setter Property="FontWeight" Value="Thin" />
+        </Style>
+        <Style x:Key="ButtonText" TargetType="Button">
+            <Setter Property="FontSize" Value="14" />
+            <Setter Property="FontFamily" Value="Segoe UI Light" />
+            <Setter Property="FontWeight" Value="Thin" />
+        </Style>
+    </Page.Resources>
```
Comment (JimGaleForce) [inline Shield/SettingsPage.xaml]: (author) added styles

File: Shield/SettingsPage.xaml [hunk 2]
```diff
@@ -47,11 +61,10 @@
         <Pivot Grid.Row="2" Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
             <PivotItem>
                 <PivotItem.Header>
-                    <TextBlock Text="Connections" FontSize="48" FontFamily="Segoe UI Light"  FontWeight="Thin" Margin="0,0,0,-18"/>
+                    <TextBlock Text="Connections" Style="{StaticResource Headers}"/>
```
Comment (JimGaleForce) [inline Shield/SettingsPage.xaml]: (author) used style (+2 other pivotitems)

File: Shield/SettingsPage.xaml [hunk 3]
```diff
@@ -90,11 +102,11 @@
...
-                                    Visibility="{Binding Source={StaticResource appSettings}, Path=ListVisible, Converter={StaticResource BoolToVisConverter}}"></Button>
+                            <Button Grid.Column="0" Height="40" x:Uid="RefreshBtn" Style="{StaticResource ButtonText}" x:Name="Refresh" Click="Refresh_Click" HorizontalAlignment="Stretch" />
+                            <Button Grid.Column="2" Height="40" x:Uid="Disconnect" Style="{StaticResource ButtonText}" x:Name="DisconncectButton" Click="Disconnect_Click" HorizontalAlignment="Stretch"
+                                    Visibility="{Binding Source={StaticResource appSettings}, Path=ListVisible, Converter={StaticResource BoolToVisConverter}}" />
+                            <Button Grid.Column="4" Height="40" x:Uid="Connect" Style="{StaticResource ButtonText}" x:Name="Reconnect" Click="Reconnect_Click" HorizontalAlignment="Stretch"
+                                    Visibility="{Binding Source={StaticResource appSettings}, Path=ListVisible, Converter={StaticResource BoolToVisConverter}}" />
```
Comment (JimGaleForce) [inline Shield/SettingsPage.xaml]: (author) used styles in buttons

File: Shield.Communication/Services/ServiceBase.cs [hunk 4]
```diff
@@ -21,85 +21,122 @@
...
+
+        private int msBetweenSends = 10;
+
+        private DateTime nextSend = DateTime.Now;
+
+        internal StreamSocket socket;
+
+        public bool IsClearToSend { get; set; }
+
+        public bool IsConnected { get; set; }
```
Comment (JimGaleForce) [inline Shield.Communication/Services/ServiceBase.cs]: (author) added isconnected for preventing disconnects

File: Shield.Communication/Services/ServiceBase.cs [hunk 5]
```diff
@@ -110,8 +147,9 @@ public virtual Task<Connections> GetConnections()
 
         public virtual void Disconnect(Connection connection)
         {
+            this.IsConnected = false;
```
Comment (JimGaleForce) [inline Shield.Communication/Services/ServiceBase.cs]: (author) modified isconnected

File: Shield.Communication/Services/ServiceBase.cs [hunk 6]
```diff
@@ -120,12 +158,13 @@ public virtual void Disconnect(Connection connection)
         {
             this.currentConnection = newConnection;
             this.OnConnect?.Invoke(newConnection);
+            this.IsConnected = true;
```
Comment (JimGaleForce) [inline Shield.Communication/Services/ServiceBase.cs]: (author) modified isconnected

File: Shield/MainPage.xaml.cs [hunk 7]
```diff
@@ -102,174 +131,198 @@ public MainPage()
...
+                    }
             }
 
-            service.OnConnect += OnConnection;
-            service.OnDisconnected += OnDisconnected;
+            this.service.OnConnect += OnConnection;
+            this.service.OnDisconnected += OnDisconnected;
+            this.service.ThreadedException += this.Service_ThreadedException;
+
+            if (!this.service.IsConnected)
```
Comment (JimGaleForce) [inline Shield/MainPage.xaml.cs]: (author) used isconnected

File: Shield/MainPage.xaml [hunk 8]
```diff
@@ -91,7 +91,7 @@
                 <RowDefinition Height="*"/>
             </Grid.RowDefinitions>
             
-            <Image Grid.Row="0" Grid.RowSpan="2" x:Name="backgroundImage"  x:FieldModifier="Internal" Height="800" Width="480"/>
+            <Image Grid.Row="0" Grid.RowSpan="2" x:Name="backgroundImage" Source="Assets/ShieldsWelcome.png"  x:FieldModifier="Internal" Height="Auto" Width="Auto" Stretch="Uniform"/>
```
Comment (JimGaleForce) [inline Shield/MainPage.xaml]: (author) new welcome graphic

File: Shield/MainPage.xaml.cs [hunk 9]
```diff
@@ -102,174 +131,198 @@ public MainPage()
...
-            };
+                {
+                    if (this.IsWelcomeMessageShowing)
+                    {
+                        this.IsWelcomeMessageShowing = false;
+                        await
+                            this.dispatcher.RunAsync(
+                                CoreDispatcherPriority.Normal, 
+                                () => { this.backgroundImage.ClearValue(Image.SourceProperty); });
+                    }
```
Comment (JimGaleForce) [inline Shield/MainPage.xaml.cs]: (author) new welcome message disappears once connected

File: Shield/MainPage.xaml.cs [hunk 10]
```diff
@@ -102,174 +131,198 @@ public MainPage()
...
+            }
 
-            service.Initialize(!appSettings.MissingBackButton);
+            this.service.ListenForBeacons();
+            this.RefreshConnections();
+        }
 
-            service.ListenForBeacons();
-            RefreshConnections();
+        private async void Service_ThreadedException(Exception exception)
```
Comment (JimGaleForce) [inline Shield/MainPage.xaml.cs]: (author) new routine to return unsupported sensor info

**Comment by sidwarkd** [comment]
👍Pulled the branch and confirmed everything builds fine. Also tested on a Nokia phone with some virtual shield samples and all worked.

**Pull request merged**